lib/src/view/relation/following_screen.dart

```diff
@@ -0,0 +1,372 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:flutter_slidable/flutter_slidable.dart';
+import 'package:lichess_mobile/src/constants.dart';
+import 'package:lichess_mobile/src/model/common/perf.dart';
+import 'package:lichess_mobile/src/model/relation/relation_ctrl.dart';
+import 'package:lichess_mobile/src/model/relation/relation_repository_providers.dart';
+import 'package:lichess_mobile/src/styles/lichess_icons.dart';
+import 'package:lichess_mobile/src/widgets/feedback.dart';
+import 'package:lichess_mobile/src/widgets/list.dart';
+
+import 'package:lichess_mobile/src/widgets/platform.dart';
+import 'package:lichess_mobile/src/utils/l10n_context.dart';
+import 'package:lichess_mobile/src/styles/lichess_colors.dart';
+import 'package:lichess_mobile/src/utils/navigation.dart';
+import 'package:lichess_mobile/src/view/user/user_screen.dart';
+import 'package:lichess_mobile/src/model/user/user.dart';
+
+final RouteObserver<PageRoute<void>> followingRouteObserver =
+    RouteObserver<PageRoute<void>>();
+
+final isInvalidFollowingCacheProvider = StateProvider<bool>((ref) => false);
+final isInvalidFollowingOnlines = StateProvider<bool>((ref) => false);
+
+class FollowingScreen extends ConsumerStatefulWidget {
+  const FollowingScreen({super.key});
+  @override
+  ConsumerState<FollowingScreen> createState() => _FollowingScreenState();
+}
+
+class _FollowingScreenState extends ConsumerState<FollowingScreen>
+    with RouteAware {
+  @override
+  Widget build(BuildContext context) {
+    return PlatformWidget(androidBuilder: _buildAndroid, iosBuilder: _buildIos);
+  }
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    final route = ModalRoute.of(context);
+    if (route != null && route is PageRoute) {
+      followingRouteObserver.subscribe(this, route);
+    }
+  }
+
+  @override
+  void dispose() {
+    followingRouteObserver.unsubscribe(this);
+    super.dispose();
+  }
+
+  @override
+  void didPop() {
+    if (ref.read(isInvalidFollowingOnlines.notifier).state) {
+      ref
+          .read(
+            relationCtrlProvider.notifier,
+          )
+          .getFollowingOnlines();
+      ref.read(isInvalidFollowingOnlines.notifier).state = false;
+    }
+    _invalidateFollowingCache();
+    super.didPop();
+  }
+
+  @override
+  void didPushNext() {
+    _invalidateFollowingCache();
+    super.didPushNext();
+  }
+
+  Widget _buildIos(BuildContext context) {
+    return const CupertinoPageScaffold(
+      navigationBar: CupertinoNavigationBar(),
+      child: _Body(),
+    );
+  }
+
+  Widget _buildAndroid(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text(context.l10n.following),
+      ),
+      body: const _Body(),
+    );
+  }
+
+  void _invalidateFollowingCache() {
```

Comment:
looks like a bit complex logic here: we should probably not cache the data at all to avoid such code

Reply:
the added benefit of caching is not worth the complexity imo